src/main/java/frc/robot/RobotContainer.java

```diff
@@ -93,46 +94,58 @@ private void configureBindings() {
                   vision.camera.setPipelineIndex(4);
                   lights.setColorHSV(259, 100, 70);
                 }));
+    controller.button(54).whileTrue(new AutoLevelingCommand(drive));
+    controller.button(31).onTrue(new GrabManipulatorCommand(manipulator, GrabObject.CONE));
+    controller.button(32).onTrue(new GrabManipulatorCommand(manipulator, GrabObject.BOX));
+    controller.button(33).onTrue(new ReleaseManipulatorCommand(manipulator));
 
     controller
         .button(3)
-        .whileTrue(Commands.startEnd(() -> armExtend.extend(), () -> armExtend.stop(), armExtend));
-    controller
-        .button(5)
-        .whileTrue(Commands.startEnd(() -> armExtend.retract(), () -> armExtend.stop(), armExtend));
+        .onTrue(
+            Commands.runOnce(
+                () -> {
+                  pivot.setGoal(Units.degreesToRadians(Constants.L2ANGLE));
+                  pivot.enable();
+                },
+                pivot));
+
     controller
         .button(4)
-        .whileTrue(Commands.startEnd(() -> armRaise.raise(), () -> armRaise.stop(), armRaise));
-    controller
-        .button(6)
-        .whileTrue(Commands.startEnd(() -> armRaise.lower(), () -> armRaise.stop(), armRaise));
-    /* controller
-        .button(1)
-        .onTrue(new ArmRaiseLowerPIDCommand(armRaise, Constants.ARM_RAISE_LOWER_SETPOINT));
-    controller
-        .button(2)
-        .onTrue(new ArmExtendRetractPIDCommand(armExtend, Constants.ARM_EXTEND_RETRACT_SETPOINT));
-    controller
-        .button(3)
-        .onTrue(new ManipulatorPIDCommand(manipulator, Constants.MANIPULATOR_SETPOINT));
-    */
-    controller.button(1).whileTrue(new AutoLevelingCommand(drive));
+        .onTrue(
+            Commands.runOnce(
+                () -> {
+                  pivot.setGoal(Units.degreesToRadians(110));
+                  pivot.enable();
+                },
+                pivot));
+
     controller
-        .button(10)
-        .whileTrue(
-            new GrabManipulatorCommand(manipulator, GrabObject.CONE)); // Button For Grabbing Cones
+        .button(5)
+        .onTrue(
+            Commands.runOnce(
+                () -> {
+                  pivot.setGoal(Constants.ARM_OFF_SET_RADS);
+                  pivot.enable();
+                },
+                pivot));
+
+    controller.button(11).whileTrue(new StartEndCommand(arm::extend, arm::stop, arm));
+    controller.button(12).whileTrue(new StartEndCommand(arm::retract, arm::stop, arm));
+
     controller
-        .button(11)
+        .button(122)
         .whileTrue(
-            new GrabManipulatorCommand(manipulator, GrabObject.BOX)); // Button For Grabbing Boxes
-    controller
-        .button(12)
-        .whileTrue(new ReleaseManipulatorCommand(manipulator)); // Button For Releasing
+            new SequentialCommandGroup(
```

Comment:
If I'm understanding this correctly, Prelevel does the initial balance and auto level keeps it in balance?
It'd be cool to stick a small comment here about that so others know

Reply:
yeah your right